crates/ironrdp-displaycontrol/src/client.rs

```diff
@@ -1,14 +1,51 @@
 use crate::{
-    pdu::{DisplayControlMonitorLayout, DisplayControlPdu, MonitorLayoutEntry},
+    pdu::{DisplayControlCapabilities, DisplayControlPdu},
     CHANNEL_NAME,
 };
 use ironrdp_dvc::{encode_dvc_messages, DvcClientProcessor, DvcMessage, DvcProcessor};
-use ironrdp_pdu::PduResult;
+use ironrdp_pdu::{cursor::ReadCursor, PduDecode, PduResult};
 use ironrdp_svc::{impl_as_any, ChannelFlags, SvcMessage};
 use tracing::debug;
 
 /// A client for the Display Control Virtual Channel.
-pub struct DisplayControlClient {}
+pub struct DisplayControlClient {
+    /// A callback that will be called when capabilities are received from the server.
+    /// If no callback is set, a default (inert) callback will be used.
+    on_capabilities_received: OnCapabilitiesReceived,
+}
+
+impl Default for DisplayControlClient {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl DisplayControlClient {
+    pub fn new() -> Self {
+        Self {
+            on_capabilities_received: Box::new(|_| {
+                debug!("No capabilities received callback set, ignoring.");
+                Ok(Vec::new())
+            }),
+        }
+    }
+
+    /// Sets a callback that will be called when capabilities are received from the server.
+    #[must_use]
+    pub fn with_capabilities_received_callback<F>(mut self, callback: F) -> Self
```

Comment:
Lets remove this and make `callback` part of `new` signature to enforce use of callback

Reply:
41f48e6ccd89b3fd87e4b9861ecbe56628db850f